specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,295 @@
+        customScheme + ":*", All);
+webView.CoreWebView2.WebResourceRequested += delegate (
+            object sender, CoreWebView2WebResourceRequestedEventArgs e) {
+        CoreWebView2WebResourceResponse response;
+        CoreWebView2WebResourceRequest request = e.Request;
+        String uri = request.uri;
+        if (uri.StartsWith(customScheme + "://"))
+        {
+            string assetsFilePath = L"data/";
+            assetsFilePath += uri.Substring(customScheme.Length + 3);
+            FileStream fileStream = new FileStream(assetsFilePath, FileMode.Open, FileAccess.Read);
+
+            if (stream)
+            {
+              e.Response = webView.CoreWebView2.CreateWebResourceResponse(fileStream, 200, "OK", L"Content-Type: application/json\nAccess-Control-Allow-Origin: *");
+            }
+            else
+            {
+                e.Response = webView.CoreWebView2.CreateWebResourceResponse(null, 404, "Not Found", L"");
+            }
+        }
+}
+
+webView.CoreWebView2.Navigate(L"https://www.example.com");
+
+// The following XHR will execute in the context of https://www.example.com page and will succeed.
+// WebResourceRequested event will be raised for this request as *.example.com is in the allowed origin list of custom-scheme.
+// Since the response header provided in WebResourceRequested handler allows all origins for CORS
+// the XHR succeeds.
+webView.CoreWebView2.ExecuteScriptAsync("var oReq = new XMLHttpRequest();\
+                                        oReq.addEventListener(\"load\", reqListener);\
+                                        oReq.open(\"GET\", \"custom-scheme://example-data.json\");\
+                                        oReq.send();");
+// The following XHR will fail because *.example.com is in the not allowed origin list of custom-scheme2.
+// The WebResourceRequested event will not be raised for this request.
+webView.CoreWebView2.ExecuteScriptAsync("var oReq = new XMLHttpRequest();\
+                                        oReq.addEventListener(\"load\", reqListener);\
+                                        oReq.open(\"GET\", \"custom-scheme-not-in-allowed-origins://example-data.json\");\
+                                        oReq.send();");
+```
+
+``` cpp
+Microsoft::WRL::ComPtr<ICoreWebView2EnvironmentOptions3> options3;
+if (options.As(&options3) == S_OK) {
+  std::vector<Microsoft::WRL::ComPtr<ICoreWebView2CustomSchemeRegistration>> schemeRegistrations;
+  
+  const WCHAR* allowedOrigins[1] = {L"https://*.example.com"};
+  schemeRegistrations.push_back(Microsoft::WRL::Make<CoreWebView2CustomSchemeRegistration>(
+          L"custom-scheme", TRUE /* isSecure*/, 1, allowedOrigins));
+  schemeRegistrations.push_back(Microsoft::WRL::Make<CoreWebView2CustomSchemeRegistration>(
+          L"custom-scheme-not-in-allowed-origins-list", TRUE /* isSecure*/, nullptr));
+  CHECK_FAILURE(options3->SetCustomSchemeRegistrations(schemeRegistrations.size(), schemeRegistrations.data());
```

Comment:
Is this the standard pattern for collection properties in the C++ variant?

Reply:
It is now for arrays.